src/index.js

```diff
@@ -28,6 +28,8 @@ const jobsReady = agenda._ready
   });
 
 const getJobMiddleware = (jobAssertion, jobOperation, errorCode = 400) => async (ctx, next) => {
+  if(ctx.request.headers['x-api-key'] != settings.appId || settings.appId == "")
```

Comment:
The CI errors you're getting are probably due to this line, think the fix is to check `settings.appId` exists first.

Reply:
I guess `settings.appId && ctx.request.headers['x-api-key'] != settings.appId` would work.